apis/core/lib/errors/validation.ts

```diff
@@ -0,0 +1,39 @@
+import { ErrorMapper } from 'http-problem-details-mapper'
+import { ProblemDocument } from 'http-problem-details'
+import { ValidationReport } from 'rdf-validate-shacl'
+
+export class ValidationError extends Error {
+  readonly validationReport: ValidationReport
+
+  constructor(validationReport: ValidationReport) {
+    super()
+
+    this.message = 'Bad request'
+    this.validationReport = validationReport
+  }
+}
+
+export class ValidationErrorMapper extends ErrorMapper {
+  private readonly status: number;
+
+  public constructor() {
+    super(ValidationError)
+    this.status = 400
```

Comment:
Is this used for anything else? Might just inline in `mapError` I suppose

Reply:
As often, I don't really understand the whole contraption we are creating. So probably not?